conversation_simulator/src/StateMachine.js

```diff
@@ -142,67 +142,74 @@ export default class {
-      } else {
-        let score = this.currentState.Solutions[solution].Score || this.currentState.Solutions[solution].scoreadjustment
-        if (this.isNumber(score)) {
-          this.scoreSolution(solution, score, noPoints)
+    if (solution === null) {
+      // Return default if it exists
+      if (this.currentState.Solutions['default']) return 'default'
+    }
+    return solution
+  }
+
+  // Returns the score of a solution
+  scoreSolution (solution, isCorrect, source) {
+    let _this = this
+    return new Promise(function (resolve, reject) {
+      dtml.scorePhrase(solution, isCorrect, result => {
+        if (result) {
+          resolve(result)
+        } else {
+          let score = _this.currentState.Solutions[solution].Score || _this.currentState.Solutions[solution].scoreadjustment
+          if (_this.isNumber(score)) {
+            resolve(score)
+          } else {
+            reject(new Error('Could not find a score'))
+          }
         }
-      }
-    }, source, this.getCurrentStateName())
+      }, source, _this.getCurrentStateName())
+    })
   }
 
-  scoreSolution (solution, score, noPoints) {
-    if (solution !== undefined) {
-      let nextState = this.currentState.Solutions[solution].Next
-      this.submitSolutionResult = true
-      if (score > 0 && !noPoints && !this.userContext.hasSolutionBeenUsed(this.getCurrentStateName(), solution)) {
-        this.score += score
-      }
-      // Save state/answer so it can't be used again for points
-      this.userContext.markSolution(this.getCurrentStateName(), solution)
-      // Go to next state
-      this.setCurrentState(
-        nextState,
-        this.stateData.States[nextState]
-      )
-    } else {
-      // If there is a default next state then it doesn't count as a failure
-      if (this.currentState.Solutions['default'].Next !== null) {
-        this.submitSolutionResult = true
-        let nextState = this.currentState.Solutions['default'].Next
-        this.setCurrentState(
-          nextState,
-          this.stateData.States[nextState]
-        )
-      } else {
-        this.submitSolutionResult = false
-        this.score -= 10
-      }
-    }
+  // Adds the score to the player's total score
+  applyScore (score) {
+    this.score += score
+  }
+
+  markSolution (stateName, solution) {
+    // Save state/answer so it can't be used again for points
+    this.userContext.markSolution(stateName, solution)
+  }
+
+  checkSolutionUsed (stateName, solution) {
+    return this.userContext.hasSolutionBeenUsed(stateName, solution)
+  }
+
+  // Goes to the state indicated by the solution
+  goToNextState (solution) {
+    let nextState = this.currentState.Solutions[solution].Next
+    this.setCurrentState(
+      nextState,
+      this.stateData.States[nextState]
+    )
   }
 
   // Recursively compare a submitted phrase vs a solution phrase
-  checkSolution (submittedWords, solutionWords) {
+  compareSolution (submittedWords, solutionWords) {
```

Comment:
This probably doesnt belong in state machine rather in some helper function

Reply:
Ok I'll move it into a separate file.